play-dl/YouTube/utils/request.ts

```diff
@@ -29,7 +30,8 @@ async function https_getter(req_url : string, options : RequestOpts = {}): Promi
 export async function request(url : string, options? : RequestOpts): Promise<string>{
     return new Promise(async (resolve, reject) => {
         let data = ''
-        let res = await https_getter(url, options)
+        let res = await https_getter(url, options).catch((err) => err as string)
+        if(typeof res === 'string') throw new Error(res)
```

Comment:
throw new Error(res) to return reject(new Error(res))
because of correct handling.